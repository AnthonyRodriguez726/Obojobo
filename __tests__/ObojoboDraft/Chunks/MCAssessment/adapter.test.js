import MCAssessmentAdapter from '../../../../ObojoboDraft/Chunks/MCAssessment/adapter'

describe('MCAssessment adapter', () => {
	it('can be constructed WITHOUT attributes', () => {
		let model = { modelState: {} }
		MCAssessmentAdapter.construct(model)
		expect(model).toMatchSnapshot()
	})

	it('can be constructed WITH attributes', () => {
		let model = { modelState: {} }
		let attrs = { content: { responseType: 'pick-one' } }

		MCAssessmentAdapter.construct(model, attrs)

		expect(model).toMatchSnapshot()
	})

	it('can be cloned', () => {
		let a = { modelState: {} }
		let b = { modelState: {} }

		MCAssessmentAdapter.construct(a)
		MCAssessmentAdapter.clone(a, b)

		expect(a).not.toBe(b)
		expect(a.modelState).toEqual(b.modelState)
	})

	it('can convert to JSON', () => {
		let model = { modelState: {} }
		let attrs = { content: { responseType: 'pick-one' } }
		let json = { content: {} }

		MCAssessmentAdapter.construct(model, attrs)
		MCAssessmentAdapter.toJSON(model, json)

		expect(json).toMatchSnapshot()
	})
})
