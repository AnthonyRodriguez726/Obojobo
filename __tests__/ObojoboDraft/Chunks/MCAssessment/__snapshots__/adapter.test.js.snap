// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MCAssessment adapter can be constructed WITH attributes 1`] = `
Object {
  "modelState": Object {
    "correctFeedbacks": Array [
      "Correct!",
      "Great job!",
    ],
    "incorrectFeedbacks": Array [
      "Incorrect",
      "wrong",
    ],
    "responseType": "pick-one",
    "shuffle": false,
  },
}
`;

exports[`MCAssessment adapter can be constructed WITH correctFeedbacks 1`] = `
Object {
  "modelState": Object {
    "correctFeedbacks": Array [
      "Correct!",
      "Great job!",
    ],
    "incorrectFeedbacks": null,
    "responseType": "",
    "shuffle": true,
  },
}
`;

exports[`MCAssessment adapter can be constructed WITH incorrectFeedbacks 1`] = `
Object {
  "modelState": Object {
    "correctFeedbacks": null,
    "incorrectFeedbacks": Array [
      "Incorrect",
      "wrong",
    ],
    "responseType": "",
    "shuffle": true,
  },
}
`;

exports[`MCAssessment adapter can be constructed WITH responseType 1`] = `
Object {
  "modelState": Object {
    "correctFeedbacks": null,
    "incorrectFeedbacks": null,
    "responseType": "pick-one",
    "shuffle": true,
  },
}
`;

exports[`MCAssessment adapter can be constructed WITH shuffle 1`] = `
Object {
  "modelState": Object {
    "correctFeedbacks": null,
    "incorrectFeedbacks": null,
    "responseType": "",
    "shuffle": false,
  },
}
`;

exports[`MCAssessment adapter can be constructed WITHOUT attributes 1`] = `
Object {
  "modelState": Object {
    "correctFeedbacks": null,
    "incorrectFeedbacks": null,
    "responseType": "",
    "shuffle": true,
  },
}
`;

exports[`MCAssessment adapter can convert to JSON 1`] = `
Object {
  "content": Object {
    "correctFeedbacks": null,
    "incorrectFeedbacks": null,
    "responseType": "pick-one",
    "shuffle": true,
  },
}
`;

exports[`MCAssessment adapter can convert to JSON WITH attributes 1`] = `
Object {
  "content": Object {
    "correctFeedbacks": "Correct!|Great job!",
    "incorrectFeedbacks": "Incorrect|wrong",
    "responseType": "pick-one",
    "shuffle": false,
  },
}
`;
